symfony/plugins/orangehrmPimPlugin/Api/EmployeeSkillAPI.php

```diff
@@ -0,0 +1,355 @@
+
+        return new EndpointUpdateResult(
+            EmployeeSkillModel::class, $employeeSkill,
+            new ParameterBag(
+                [
+                    CommonParams::PARAMETER_EMP_NUMBER => $employeeSkill->getEmployee()->getEmpNumber(),
+                    self::PARAMETER_SKILL_ID => $employeeSkill->getSkill()->getId()
+                ]
+            )
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForUpdate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(CommonParams::PARAMETER_ID, new Rule(Rules::REQUIRED)),
+            $this->getEmpNumberRule(),
+            ...$this->getCommonBodyValidationRules(),
+        );
+    }
+
+    /**
+     * @inheritDoc
+     * @throws DaoException
+     * @throws Exception
+     */
+    public function delete(): EndpointDeleteResult
+    {
+        $empNumber = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            CommonParams::PARAMETER_EMP_NUMBER
+        );
+        $ids = $this->getRequestParams()->getArray(RequestParams::PARAM_TYPE_BODY, CommonParams::PARAMETER_IDS);
+        $this->getEmployeeSkillService()->getEmployeeSkillDao()->deleteEmployeeSkills($empNumber, $ids);
+        return new EndpointDeleteResult(
+            ArrayModel::class, $ids,
+            new ParameterBag(
+                [
+                    CommonParams::PARAMETER_EMP_NUMBER => $empNumber,
+                ]
+            )
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForDelete(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getEmpNumberRule(),
+            new ParamRule(CommonParams::PARAMETER_IDS),
+        );
+    }
+
+    /**
+     * @return EmployeeSkill
+     * @throws RecordNotFoundException|DaoException
+     */
+    public function saveEmployeeSkill(): EmployeeSkill
+    {
+        $id = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, CommonParams::PARAMETER_ID);
+        $skillId = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_SKILL_ID);
+        $comments = $this->getRequestParams()->getStringOrNull(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_COMMENTS
+        );
+        $yrsOfExp = $this->getRequestParams()->getIntOrNull(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_YEARS_OF_EXP
+        );
+        $empNumber = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            CommonParams::PARAMETER_EMP_NUMBER
+        );
+
+        // update operation
+        if (!empty($id)) {
+            $employeeSkill = $this->getEmployeeSkillService()->getEmployeeSkillDao()->getEmployeeSkillById(
+                $empNumber,
+                $id
+            );
+            if ($employeeSkill == null) {
```

Comment:
remove if condition, just use `$this->throwRecordNotFoundExceptionIfNotExist($employeeSkill, EmployeeSkill::class);`